ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -406,6 +496,78 @@ def test_delete_deps_for_rebuilding_hostgroups(self, host1, hostgroup1,
         automember_hostgroup.delete()
 
 
+class TestAutomemberModifyNegativeSameval(XMLRPC_test):
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_modify_with_same_value(self, automember_grp, grp, request):
+        """ Try to modify an existing rule with the same value """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_exists()
+        automember_grp.ensure_exists()
+        command = automember_grp.make_update_command(updates=dict(
+            description='%s' % automember_grp.description))
+        with pytest.raises(errors.EmptyModlist,
+                           match=r"no modifications to be performed"):
+            command()
+
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_modify_with_nonexistent_group(self, automember_grp, grp, request):
+        """ Try to modify a rule with non-existent group/hostgroup """
+        grp = request.getfixturevalue(grp)
```

Comment:
The function can be simplified and use only automember_grp + request parameters (grp is not useful here, grp.cn can be replaced with automember_grp.cn).